### PR TITLE
Fix make cmds

### PR DESCRIPTION
**Pull request by fabianburth**
#### What this PR does / why we need it:
This PR adds `gci` to `make install-requirements`. This fixes a bug in the `gci write` commands which lead to the `slices` import being wrongly sorted which made `make format` and also `make prepare` essentially unusable. 

During the adjustment, I also changed `make test-all` to `make test` and the previous `make test` to `make unit-test`. I think the default should be to run all tests (currently, they dont take too long anyway).

### CODE REVIEW

File: hack/Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ ifeq ("v$(JQ)","v0.0")
 	deps += $(detected_OS)_jq
 endif
 
-GOLANGCILINT_VERSION := "v1.54.2"
+GOLANGCILINT_VERSION := "v1.59.1"
```
Comment (Skarlso) [inline hack/Makefile:36]: This will result in a f*ckton of errors. :D Just FYI :D

File: Makefile [hunk 2]
```diff
@@ -54,6 +54,10 @@ format:
 check:
 	@$(REPO_ROOT)/hack/check.sh --golangci-lint-config=./.golangci.yaml $(REPO_ROOT)/cmds/ocm $(REPO_ROOT)/cmds/helminstaller/... $(REPO_ROOT)/cmds/ecrplugin/... $(REPO_ROOT)/cmds/demoplugin/... $(REPO_ROOT)/pkg/...
 
+.PHONY: check-and-fix
+check-with-fix:
+	@$(REPO_ROOT)/hack/check.sh --fix --golangci-lint-config=./.golangci.yaml $(REPO_ROOT)/cmds/ocm $(REPO_ROOT)/cmds/helminstaller/... $(REPO_ROOT)/cmds/ecrplugin/... $(REPO_ROOT)/cmds/demoplugin/... $(REPO_ROOT)/pkg/...
```
Comment (hilmarf) [inline Makefile:59]: why aren't the three dots (`/...`) at ocm command required?

```suggestion
	@$(REPO_ROOT)/hack/check.sh --fix --golangci-lint-config=./.golangci.yaml $(REPO_ROOT)/cmds/ocm/... $(REPO_ROOT)/cmds/helminstaller/... $(REPO_ROOT)/cmds/ecrplugin/... $(REPO_ROOT)/cmds/demoplugin/... $(REPO_ROOT)/pkg/...
```

Comment (fabianburth) [reply Makefile]: Good catch! Those files have essentially never been linted. Had to fix quite some errors.

File: cmds/ocm/pkg/processing/logging.go [hunk 3]
```diff
@@ -6,4 +6,4 @@ import (
 
 var REALM = ocmlog.DefineSubRealm("output processing chains", "processing")
 
-var logger = ocmlog.DynamicLogger()
+// var logger = ocmlog.DynamicLogger()
```
Comment (hilmarf) [inline cmds/ocm/pkg/processing/logging.go:9]: shouldn't it be removed completely?

Comment (fabianburth) [reply cmds/ocm/pkg/processing/logging.go]: Done

**Review by hilmarf — APPROVED** [review]

**Review by mandelsoft — DISMISSED** [review]

**Review by mandelsoft — APPROVED** [review]

**Pull request merged**